server/src/test/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportBootstrapClusterActionTests.java

```diff
@@ -117,7 +117,7 @@ public void handleResponse(BootstrapClusterResponse response) {
             public void handleException(TransportException exp) {
                 final Throwable rootCause = exp.getRootCause();
                 assertThat(rootCause, instanceOf(IllegalArgumentException.class));
-                assertThat(rootCause.getMessage(), equalTo("cluster bootstrapping is not supported by discovery type [zen]"));
+                assertThat(rootCause.getMessage(), equalTo("cluster bootstrapping is not supported by discovery type [zen2]"));
```

Comment:
Um, cluster bootstrapping _is only_ supported by discovery type `zen2` so this seems wrong. I think the right fix is to pass another discovery type into the `TransportBootstrapClusterAction` constructor above, choosing randomly from `zen`, `single-node` and some other random string.